lib/querystring.js

```diff
@@ -170,6 +189,8 @@ function stringify(obj, sep, eq, options) {
   if (options && typeof options.encodeURIComponent === 'function') {
     encode = options.encodeURIComponent;
   }
+  const convert =
+    (encode === qsEscape ? encodeStringified : encodeStringifiedCustom);
```

Comment:
Please add a test that verifies that it would break in case the user would manipulate encode.

```js
const querystring = require('querystring');
querystring.encode = (arg) => {
  assert.strictEqual(typeof arg, 'string');
};
querystring.stringify({ bool: true });
```

Reply:
I'm not sure what you're asking for that isn't already covered by existing tests?

Reply:
ping @BridgeAR